### PR TITLE
WIP: Fix time across composed log sites

### PR DESCRIPTION
**Pull request by jcorbin**
Currently with a zap.Tee logger:
- each sub logger calls time.Now itself at moderately different times
- so does every (*CheckedMessage).Write in a chain

The approach here is to pass a single `time.Time` along to `Logger.Log`.

I can easily separate the entry refactor work (it's already in a separate branch `imp_entry`) if desired.

### CODE REVIEW

File: checked_message.go [hunk 1]
```diff
@@ -83,10 +91,12 @@ func (m *CheckedMessage) Write(fields ...Field) {
 	case FatalLevel:
 		m.logger.Fatal(m.msg, fields...)
 	default:
-		m.logger.Log(m.lvl, m.msg, fields...)
+		m.logger.Log(t, m.lvl, m.msg, fields...)
```
Comment (akshayjshah) [inline checked_message.go]: Can we leave a TODO to use `logger.Log` and propagate time in the above cases too? Probably best to address that when we make it easier to write new loggers (i.e., when we're 100% sure that all `logger.Panic` methods just call `logger.Log` and the `panic`).

**Comment by jcorbin** [comment]
Tabling this PR for now, as it has reveal need for an interface reworking to manage the emerging complexity of logger / message composition.

**Comment by akshayjshah** [comment]
This is now part of #201, right? If so, can we close this one?

**Comment by jcorbin** [comment]
Right, this one is superceded.

**Pull request closed without merging**